app/core/src/main/java/com/fsck/k9/helper/Utility.java

```diff
@@ -312,6 +328,95 @@ public static String extractMessageId(final String text) {
         return null;
     }
 
+    public static String getResizedImageFile(Context context, Uri uri, int circumference, int quality) {
+        File cacheDir = context.getCacheDir();
+        File tempAttachmentsDirectory = new File(cacheDir.getPath() + RESIZED_IMAGES_TEMPORARY_DIRECTORY);
+        tempAttachmentsDirectory.mkdirs();
+
+        File tempFile;
+        Bitmap bitmap = null;
+        Bitmap resized;
+        FileOutputStream out = null;
+
+        float factor = 0;
+
+        try {
+            tempFile = File.createTempFile("TempResizedAttachment", null, tempAttachmentsDirectory);
+            bitmap = MediaStore.Images.Media.getBitmap(context.getContentResolver(), uri);
+
+            factor = (bitmap.getWidth() + bitmap.getHeight() + 0f) / circumference;
+            int newWidth;
+            int newHeight;
+            if (factor <= 1.0f) {
+                newWidth = bitmap.getWidth();
+                newHeight = bitmap.getHeight();
+            } else {
+                newWidth = (int) (bitmap.getWidth() / factor);
+                newHeight = (int) (bitmap.getHeight() / factor);
+
+                while (newWidth + newHeight < circumference) {
+                    if ((0f + bitmap.getWidth()) / newWidth >= (0f + bitmap.getHeight()) / newHeight)
+                        ++newWidth;
+                    else
+                        ++newHeight;
+                }
+            }
+
+            resized = Bitmap.createScaledBitmap(bitmap, newWidth, newHeight, true);
+            out = new FileOutputStream(tempFile);
+            resized.compress(Bitmap.CompressFormat.JPEG, quality, out);
+        } catch (IOException | OutOfMemoryError e) {
+            Timber.e(e, "Error while resizing image attachment: (" + bitmap.getWidth() + "," + bitmap.getHeight() + ") / " + factor);
+            return "";
```

Comment:
Returning an empty string when resizing fails feels very wrong. Why not throw the IOException?